### DISCUSSION TITLE
The programmable card docs are missing the afterDecline function

### DISCUSSION THREAD

**Discussion by devinpearson** [Investec-Developer-Community/.github#14] [Bugs]
If you navigate to the [Programmable card](https://developer.investec.com/za/programmable-card) docs, you will only see the beforeTransaction and the afterTransaction, but there is no mention of the afterDecline.

afterDecline should be added to the docs and the examples.

**Comment by devinpearson** [comment]
Looking again it is mentioned. It can be explained a little further and shown in the examples. 

None of the functions are required to be present in the code the user submits. If they only want to act on card declines, they can supply only the afterDecline.

**Comment by petersmythe** [comment]
I agree that the `afterDecline` stub should be added to the docs, examples/tutorials, and especially the Reset code to default button:

![image](https://github.com/Investec-Developer-Community/.github/assets/5699667/1d497278-7915-4ba7-8b45-3c97d1b2b930)

If you don't supply it, any decline generates an error message: 

```json
{
    "executionId": "2EE2F5F0-7090-4B8C-",
    "rootCodeFunctionId": "45544A34-3805-",
    "sandbox": false,
    "type": "after_decline",
    "authorizationApproved": null,
    "logs": [
        {
            "createdAt": "2024-05-01T10:52:58.764Z",
            "level": "error",
            "content": "Error: afterDecline is not defined"
        }
    ],
    "smsCount": 0,
    "emailCount": 0,
    "pushNotificationCount": 0,
    "createdAt": "2024-05-01T10:52:58.285Z",
    "startedAt": "2024-05-01T10:52:58.285Z",
    "completedAt": "2024-05-01T10:52:58.901Z",
    "updatedAt": "2024-05-01T10:52:58.285Z"
}
```
